NeoML/src/TraditionalML/SparseFloatMatrix.cpp

```diff
@@ -182,8 +216,10 @@ void CSparseFloatMatrix::AddRow( const CFloatVectorDesc& row )
 		}
 	}
 
+	NeoAssert( body->Desc.Height <= MaxBufferSize - 1 );
 	GrowInRows( body->Desc.Height + 1 );
 	if( size > 0 ) {
+		NeoAssert( body->ElementCount <= MaxBufferSize - size );
 		GrowInElements( body->ElementCount + size );
```

Comment:
Double allocation of element buffers in case of RecCount() != 1.

Reply:
fixed